types/tx/signing/signature.go

```
@@ -66,8 +71,13 @@
 func SignatureDataFromProto(descData *SignatureDescriptor_Data) SignatureData {
 	switch descData := descData.Sum.(type) {
 	case *SignatureDescriptor_Data_Single_:
+		signMode, err := InternalSignModeToAPI(descData.Single.Mode)
+		if err != nil {
+			panic(fmt.Errorf("invalid sign mode: %v", err))
```

golangci-lint (1.62.2)

non-wrapping format verb for fmt.Errorf. Use `%w` to format errors

(errorlint)